### PR TITLE
Add node labels to kubelet

### LINKED ISSUE #46: labels do not get applied to nodes
Update the kubelet configuration to apply the flags / config with data from this: https://github.com/UKHomeOffice/keto/issues/6

### PR DESCRIPTION
**Pull request by lewismarshall**
Fixes #46.

Dependant on https://github.com/UKHomeOffice/keto/pull/80

### CODE REVIEW

File: pkg/kmm/kubelet.go [hunk 1]
```diff
@@ -17,15 +17,26 @@ import (
 // CreateAndStartKubelet will call the CreateAndStartKubelet method with the correct configuration
 func (k *Kmm) CreateAndStartKubelet(master bool) error {
 
+	nodeLables := ""
+	if k.NodeLabels != nil {
+		for k, v := range k.NodeLabels {
+			if len(nodeLables) > 0 {
+				nodeLables = nodeLables + ","
```
Comment (gambol99) [inline pkg/kmm/kubelet.go]: would this not place a ',' at the end of the line? .. perhaps just add a list a strings.Join with a comma

File: pkg/kmm/kmm.go [hunk 2]
```diff
@@ -285,15 +286,12 @@ func (k *Kmm) UpdateCloudCfg() (err error) {
...
-		if clusterName, err = node.GetClusterName(); err != nil {
-			return fmt.Errorf("Error getting cluster name cloud provider:%q", err)
-		}
-		k.ClusterName = clusterName
-		var api string
-		if api, err = node.GetKubeAPIURL(); err != nil {
-			return fmt.Errorf("Error getting Api server from cloud provider:%q", err)
+		nd, err := node.GetNodeData()
+		if err != nil {
+			return fmt.Errorf("Error getting node data from cloud provider:%q", err)
```
Comment (gambol99) [inline pkg/kmm/kmm.go]: a space after the ':'

File: pkg/kmm/kubelet.go [hunk 3]
```diff
@@ -17,15 +17,26 @@ import (
...
+			}
+			nodeLables = nodeLables + k + "=" + v
```
Comment (lewismarshall) [inline pkg/kmm/kubelet.go]: @gambol99 - as this is done afterwards the comma will never end the line - but I agree - I'll push a join 👍

Comment (lewismarshall) [inline pkg/kmm/kubelet.go]: Actually, it's a map - need to Join expects an array - so nogo!

File: pkg/kmm/kmm.go (hunk 2 above)
Comment (vaijab) [inline pkg/kmm/kmm.go]: Also error messages should always start with a lowercase char by convention.

File: pkg/kmm/kubelet.go (hunk 3 above)
Comment (vaijab) [inline pkg/kmm/kubelet.go]: Could do like this: https://github.com/UKHomeOffice/keto/blob/master/pkg/keto/resource_printer.go#L96

**Comment by KashifSaadat** [comment]
Building on `master` branch of keto (which has PR#80 merged in) and referencing the latest quay build for the labels branch (`153e2c23af8b5eb7094d158cfcd298672ca8e40a`).

The master nodes fail to retrieve cert files, causing the `keto-k8` service to fail:

```
Jul 05 15:37:16 ip-172-30-1-250.eu-west-1.compute.internal systemd[1]: Starting Keto K8 Service...
Jul 05 15:37:16 ip-172-30-1-250.eu-west-1.compute.internal cp[996]: /usr/bin/cp: cannot stat '/data/ca/etcd/ca.crt': No such file or directory
Jul 05 15:37:16 ip-172-30-1-250.eu-west-1.compute.internal systemd[1]: keto-k8.service: Control process exited, code=exited status=1
Jul 05 15:37:16 ip-172-30-1-250.eu-west-1.compute.internal systemd[1]: Failed to start Keto K8 Service.
Jul 05 15:37:16 ip-172-30-1-250.eu-west-1.compute.internal systemd[1]: keto-k8.service: Unit entered failed state.
Jul 05 15:37:16 ip-172-30-1-250.eu-west-1.compute.internal systemd[1]: keto-k8.service: Failed with result 'exit-code'.
Jul 05 15:37:37 ip-172-30-1-250.eu-west-1.compute.internal systemd[1]: keto-k8.service: Service hold-off time over, scheduling restart.
Jul 05 15:37:37 ip-172-30-1-250.eu-west-1.compute.internal systemd[1]: Stopped Keto K8 Service.
```

This is in a continuous loop, no success.

**Comment by lewismarshall** [comment]
As discussed @KashifSaadat it's the etcd-member service with `kmm save-assets` command that's failing due to no e2e :).

I've created new issue: #49

**Pull request merged**